aspects/schema.go

```diff
@@ -90,14 +94,40 @@ func ParseSchema(raw []byte) (*StorageSchema, error) {
 	return schema, nil
 }
 
+// userTypeRefParser parses references to user-defined types (e.g., $my-type).
+type userTypeRefParser struct {
+	parser
+
+	stringBased bool
+}
+
+func newUserTypeRefParser(p parser) *userTypeRefParser {
+	_, ok := p.(*stringSchema)
```

Comment:
it's sligthly slower but maybe simpler to do this check directly in isStringBased itself?